templates/user/edit-profile.html

```diff
@@ -262,6 +357,24 @@
                                 {{ _('Change your password') }}
                             </a>
                         </td></tr>
+                        {% if profile.is_totp_enabled %}
+                        <tr>
+                            <td colspan="2"><span class="inline-header grayed">
+                                {{ _('Scratch Codes:') }}
+                                {% if profile.scratch_codes %}
+                                    <a id="generate-scratch-codes-button" href="{{ url('generate_scratch_codes') }}"
+                                       class="button inline-button">{{ _('Regenerate') }}</a>
+                                    <div class="scratch-codes-text" id="scratch-codes-text">{{ _('Hidden') }}</div>
+                                    <pre><code id="scratch-codes"></code></pre>
+                                {% else %}
+                                    <a id="generate-scratch-codes-button" href="{{ url('generate_scratch_codes') }}"
+                                       class="button inline-button">{{ _('Generate') }}</a>
+                                    <div class="scratch-codes-text" id="scratch-codes-text"></div>
```

Comment:
I don't see the reason for having both the class and id to be `scratch-codes-text` here... either would work, but only one is needed.

Reply:
One is used for the text that appears after the `generate/regenerate` button is pressed and the other is to match the font used for `Hidden`.